### PR TITLE
Add com.zquestclassic.ZQuest

### PR DESCRIPTION
**Pull request by connorjclark**
This is for the game engine ZQuest Classic. Some important resources are:

- The repo: https://github.com/ArmageddonGames/ZQuestClassic/
- The game database / main community: https://www.purezc.net/
- The homepage: https://zeldaclassic.com/ (but soon to be https://zquestclassic.com/, which is why that is our app id)

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge). (Note: I _do_ own `zquestclassic.com`, but we haven't moved from the older domain yet, hence our homepage here is zeldaclassic.com)
- [x] (**N/A**) Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: com.zquestclassic.ZQuest.metainfo.xml
Comment (barthalion) [inline com.zquestclassic.ZQuest.metainfo.xml:1]: Ideally, you should maintain this file in your repo and include it in the release.

File: com.zquestclassic.ZQuest.desktop
Comment (barthalion) [inline com.zquestclassic.ZQuest.desktop:1]: Ideally, you should maintain this file in your repo and include it in the release.

File: com.zquestclassic.ZQuest.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
+app-id: com.zquestclassic.ZQuest
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+add-extensions:
+  org.gnome.Platform.Compat.i386:
+    directory: lib/i386-linux-gnu
+    version: '44'
```
Comment (barthalion) [inline com.zquestclassic.ZQuest.yml:8]: This seems wrong. Are you sure it shouldn't be org.freedesktop.Platform?

File: com.zquestclassic.ZQuest.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+finish-args:
+  - --device=all
+  - --share=network
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --filesystem=xdg-download
+  - --filesystem=xdg-documents
+  - --filesystem=xdg-data/ZQuest Classic:create
```
Comment (barthalion) [inline com.zquestclassic.ZQuest.yml:17]: Is it for backwards compatibility? Applications have access to $XDG_DATA_HOME by default, as long as they actually use that variable.

File: com.zquestclassic.ZQuest.yml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      - type: file
+        path: icons/com.zquestclassic.ZQuest-64.png
+      - type: file
+        path: icons/com.zquestclassic.ZQuest-128.png
+      - type: file
+        path: icons/com.zquestclassic.ZQuest-192.png
+      - type: file
+        path: icons/com.zquestclassic.ZQuest-256.png
+      - type: file
+        path: icons/com.zquestclassic.ZQuest-512.png
```
Comment (barthalion) [inline com.zquestclassic.ZQuest.yml:113]: Just the 512x512 variant is enough if it's not used internally by the app.

File: com.zquestclassic.ZQuest.yml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  - --socket=x11
+  - --socket=pulseaudio
+  - --filesystem=xdg-download
+  - --filesystem=xdg-documents
+  - --filesystem=xdg-data/ZQuest Classic:create
+  # Needed for game controllers.
+  - --device=all
+command: entry.sh
+build-options:
+  no-debuginfo: true
```
Comment (barthalion) [inline com.zquestclassic.ZQuest.yml:22]: What's the story behind this?

File: com.zquestclassic.ZQuest.yml [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+          # For now, just copy all the base resources to the user data dir,
+          # and run the program from there.
+          - |
+            set -ex
+            DATA_DIR="${XDG_DATA_HOME:-$HOME/.local/share}"
+            mkdir -p "$DATA_DIR/ZQuest Classic"
+            cp -r /app/resources/* "$DATA_DIR/ZQuest Classic"
+          - |
+            cd "$DATA_DIR/ZQuest Classic"
+            PATH=/app/bin:$PATH zlauncher
```
Comment (hfiguiere) [inline com.zquestclassic.ZQuest.yml:90]: `/app/bin` is always in the path.

File: com.zquestclassic.ZQuest.yml [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+        find . -name "*.so*" -exec mv {} /app/bin/ \;
+        find . -perm /a+x -type f -exec mv {} /app/bin/ \;
+        rm com.zquestclassic.*
+        mkdir /app/resources/
+        mv * /app/resources/
+
+    sources:
+      - type: archive
+        url: https://github.com/ArmageddonGames/ZQuestClassic/releases/download/nightly-2023-07-19/nightly-2023-07-19-linux.tar.gz
```
Comment (hfiguiere) [inline com.zquestclassic.ZQuest.yml:68]: why not building it from source and have an aarch64 package?

File: com.zquestclassic.ZQuest.yml [hunk 7]
```diff
@@ -0,0 +1,109 @@
+app-id: com.zquestclassic.ZQuest
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+finish-args:
+  - --device=all
```
Comment (hfiguiere) [inline com.zquestclassic.ZQuest.yml:6]:
```suggestion
```

you don't need it twice

File: com.zquestclassic.ZQuest.yml (hunk 3 above)
Comment (barthalion) [reply com.zquestclassic.ZQuest.yml:105]: Anything above 128x128 will be downscaled to these two sizes, but will also look nicer on the actual desktop. 192x192 is also fine if that's the source image; feel free to change that after the merge.

**Comment by barthalion** [comment]
bot, build com.zquestclassic.ZQuest

File: com.zquestclassic.ZQuest.metainfo.xml
Comment (barthalion) [reply com.zquestclassic.ZQuest.metainfo.xml:1]: No, you would need to keep it updated manually. It's not a blocker, maybe you could address this when there's a stable build.

**Pull request closed without merging**